File: N-Repeated Element in Size 2N Array/Solution.rb
# @param {Integer[]} a
# @return {Integer}
def repeated_n_times(a)
    occur = Hash.new(0)
    n =  a.length/2
    a.each {|i|
        occur[i] +=1
    }
    occur.each {|k,v|
        return k if v==n
    } 
end

File: N-Repeated Element in Size 2N Array/Solution.java
class Solution {
    public int repeatedNTimes(int[] A) {
        HashMap<Integer,Integer> occur = new HashMap<>();
        int N = A.length/2;
        int test = -1;
        for(int i : A){
            if(occur.containsKey(i)){
                occur.put(i,occur.get(i)+1);
            }else{
                occur.put(i,1);
            }
        }
        Iterator<HashMap.Entry<Integer, Integer>> iter = occur.entrySet().iterator();
        while (iter.hasNext()) {
            HashMap.Entry<Integer, Integer> kv = iter.next();
            if(kv.getValue()==N) return kv.getKey();     
        }
        return -1;
    }
}